### DISCUSSION TITLE
How to access store in a websocket route handler?

### DISCUSSION THREAD

**Discussion by Malivuca** [elysiajs/elysia#581] [Q&A]
It is easy to access the global `store` inside the GET, POST, PUT, DELETE route handlers. As from the v1.0 docs:
```ts
import { Elysia } from 'elysia'

new Elysia()
    .state('version', 1)
    .get('/a', ({ store: { version } }) => version) // easily having access to the store object
    .get('/b', ({ store }) => store)  // also works fine
    .listen(3000)
```
But how could I access the `store` in a websocket route?
```ts
import { Elysia } from 'elysia'

new Elysia()
    .ws('/ws', 
     // --------How can I get the store value here?--------
    {
        message(ws, message) {
            ws.send(message)
        }
    })
    .listen(3000)
```
Also, I want to be able to access and update the store properties inside the `open()` callback, after doing some operations, like this:
```ts
import { Elysia } from 'elysia'

new Elysia()
    .ws('/ws', 
     {
         open(ws) {
           // do some operations
           store.myProperty = true // want to update store value here, but how?
         }
      }
    })
    .listen(3000)
```

**Comment by devcayed** [comment]
Has there been any update on this?

**Comment by m50S79sM6SRNp8Jn** [comment]
>         message(ws, message) {
>             ws.send(message)
>         }
>     })

‘ message(ws, message) {
      const version = ws.data.store.version
      ws.send(`Echo: ${message}, Version: ${version}`)
    }
  }’
